### PR TITLE
Register could be a not qualified name

### PR DESCRIPTION
**Pull request by bougar**
- I modified the code to suppor non qualified register names.
- Throws error if image is not found

### CODE REVIEW

**Comment by hashmap** [comment]
@bougar thanks for your contribution! would you mind to add a test case?

File: docker/docker.go [hunk 1]
```diff
@@ -116,9 +116,11 @@ func NewImage(qname, user, password string, insecureTLS, insecureRegistry bool)
 			}
 			part := qname[start:i]
 			start = i + 1
+
 			switch state {
 			case stateInitial:
-				if part == "localhost" || strings.Contains(part, ".") {
+                if part == "localhost" || strings.Contains(part, ".") || string(qname[i]) == ":" {
```
Comment (supereagle) [inline docker/docker.go]: Need `go fmt`.

**Comment by bougar** [comment]
Test case added and fixed code to work with no_registry_root case

**Comment by hashmap** [comment]
Thanks! Correct me if I wrong - this fix implies that port must be always specified? Like `registry/nginx` is not supported.

**Pull request merged**